### ISSUE TITLE
More advanced joins 

### ISSUE THREAD

**Issue by jmatsushita** [beautifulinteractions/node-quadstore#41]
Hi there,

Currently I believe it's only possible to join two queries by an identical property (i.e. `predicate` to `predicate`)

It would be great to be able to join queries by joining on different fields. This kind of API would do I guess:
```
// QuadStore
const matchTermsA = {graph: 'g'};
const matchTermsB = {subject: 's'};
quadStore.query(matchTermsA)
    .join(quadStore.query(matchTermsB), [{ 'subject': 'graph' }]) // this would join (rather arbitrarily) a.graph == b.subject)  
    .get((err, quads) => {});
```

Allowing to still pass strings would keep backward compatibility, and allowing objects would enable joining by different properties.

What do you think?

Jun

**Comment by jacoscaz** [comment]
Hello @jmatsushita. This is a fantastic suggestion. I have a few hours of spare time coming up. I'll look into this and #40 .

**Comment by jmatsushita** [comment]
Great ! :)

**Comment by jacoscaz** [comment]
Some preliminary work about this here: https://github.com/jacoscaz/node-quadstore/blob/joins/test/quadstore.js#L586

I've had to slightly changed your proposal as I needed to maintain the ordering of term names:

```
quadStore.query(matchTermsA)
    .join(quadStore.query(matchTermsB), ['subject'], ['graph'])
    .get((err, quads) => {});
```

**Comment by jacoscaz** [comment]
See pull request #43

**Comment by jacoscaz** [comment]
Closing, available on NPM in quadstore@2.2.0-alpha .
